### PR TITLE
added circularLinkedList

### PR DESCRIPTION
**Pull request by reethikprasad**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Added Circular LinkedList 
>Creation, Insertion, getting data to be deleted, deletion

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Comment by reethikprasad** [comment]
I have added the C++ code as a part of Hackins2020

**Comment by reethikprasad** [comment]
@HarshCasper I have solved merge conflict can you please merge it to the master branch

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@reethikprasad Please do following modifications
* For friendly user interface, take input from the user & display appropriate message after each operation
* Create a count function for counting the total number of nodes present at that time.
* Create a exit function, for exiting out from the program
* Also remove the **merge conflicts** error

File: C-Plus-Plus/ds/Circular_LinkedList.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    //self loop for single node
+    else
+    {
+        n->next = n;
+    }
+    head = n;
+}
+
+//Creating a function the get the node to be deleted
+node *getnode(node *head, int data) //pass ny value
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/ds/Circular_LinkedList.cpp:41]: correct typo in comment

File: C-Plus-Plus/ds/Circular_LinkedList.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        cout << temp->data << " -> ";
+        temp = temp->next;
+    }
+    cout << temp->data << endl; //for last node
+}
+
+int main()
+{
+    node *head = NULL;
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/ds/Circular_LinkedList.cpp:98]: dynamic input

File: C-Plus-Plus/ds/Circular_LinkedList.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    insert(head, 1);
+    insert(head, 2);
+    insert(head, 3);
+    insert(head, 4);
+    insert(head, 5);
+    insert(head, 6);
+    print(head);
+
+    del(head, 4);
+    del(head, 5);
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/ds/Circular_LinkedList.cpp]: print an appropriate message for every output

**Review by vatsalkesarwani12 — COMMENTED** [review]
Do the changes

**Comment by atarax665** [comment]
@reethikprasad  Do the suggested changes ASAP.

**Comment by reethikprasad** [comment]
On it 😅

File: C-Plus-Plus/ds/Circular_LinkedList.cpp (hunk 1 above)
Comment (reethikprasad) [reply C-Plus-Plus/ds/Circular_LinkedList.cpp:41]: Done

File: C-Plus-Plus/ds/Circular_LinkedList.cpp (hunk 2 above)
Comment (reethikprasad) [reply C-Plus-Plus/ds/Circular_LinkedList.cpp:98]: done

File: C-Plus-Plus/ds/Circular_LinkedList.cpp (hunk 3 above)
Comment (reethikprasad) [reply C-Plus-Plus/ds/Circular_LinkedList.cpp]: done

**Comment by reethikprasad** [comment]
I have made the requested changes on my branch, please check @siddharth25pandey @iamrajiv

File: C-Plus-Plus/ds/Circular_LinkedList.cpp (hunk 2 above)
Comment (reethikprasad) [reply C-Plus-Plus/ds/Circular_LinkedList.cpp:98]: @plazzy99

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @reethikprasad for code

**Pull request merged**